### ISSUE TITLE
git update-index --no-assume-unchanged not working with absolute paths on SUBST drive

### ISSUE THREAD

**Issue by quasarea** [git-for-windows/git#3908]
- [v] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
```
$ git --version --build-options

git version 2.36.1.windows.1
cpu: x86_64
built from commit: e2ff68a2d1426758c78d023f863bfa1e03cbc768
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19042.1706]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
> type "$env:USERPROFILE\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VisualStudioCode
Custom Editor Path:
Default Branch Option: main
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Rebase
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Enabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

No

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
subst z: c:\myGitRepo
cd /d z:
git update-index --no-assume-unchanged z:\my-file.txt
```
 - What did you expect to occur after running these commands?

nothing, no error, no output

 - What actually happened instead?
```
fatal: 'z:\my-file.txt' is outside repository at 'C:/myGitRepo'
```

using a relative path does work fine, but makes using the command much more  complicated in my case

**Comment by dscho** [comment]
The `update-index --no-assume-unchanged` part is probably unrelated to this. The underlying cause is probably that Git does not expect a non-canonicalized path.

Couldn't you use `git update-index -C Z:\ update-index --no-assume-unchanged my-file.txt`? That would not have to deal with non-canonicalized paths at all.

**Comment by quasarea** [comment]
Unfotunately doing 
```
git -C Z:\ update-index --assume-unchanged Z:\nuget.config
```
yields
```
fatal: 'Z:\nuget.config' is outside repository at 'D:/azagent/A1/_work/31/s/repo'
```

**Comment by dscho** [comment]
> Unfotunately doing 
> ```
> git -C Z:\ update-index --assume-unchanged Z:\nuget.config
> ```
> yields
> ```
> fatal: 'Z:\nuget.config' is outside repository at 'D:/azagent/A1/_work/31/s/repo'
> ```

No, I really meant specifying a _relative_ path for the file, _not_ an absolute one. I.e. `nuget.config` instead of `Z:\nuget.config`.

**Comment by mfriedrich74** [comment]
Subst drives by default are visible only to the user session that created
them. So, if it was created without elevation it's not usable for scripts
running elevated and vice versa. Unless a special registry key is set to
allow sharing. But there are scenarios where even that doest work. Subst
are definitely not visible to other users, if a user change is at play here.

So, my general advice is to avoid using subst as much as possible.
I hope this info helps analyzing further.

Best regards, Mike
